### PR TITLE
Add channel picker in media_player dialog

### PR DESCRIPTION
**Pull request by stephanerosi**
Allows choosing a channel in a media_player component that supports it

image

Need this PR in the backend : https://github.com/home-assistant/home-assistant/pull/15941

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-media_player.js [hunk 1]
```diff
@@ -255,6 +280,27 @@ class MoreInfoMediaPlayer extends LocalizeMixin(EventsMixin(PolymerElement)) {
     this.playerObj.selectSource(sourceInput);
   }
 
+  handleChannelChanged(channelIndex, channelIndexOld) {
+    // Selected Option will transition to '' before transitioning to new value
+    if (!this.playerObj
+        || this.playerObj.channelList === undefined
+        || channelIndex < 0
+        || channelIndex >= this.playerObj.channelList
```
Comment (jeradM) [inline src/dialogs/more-info/controls/more-info-media_player.js]: Comparing number to array? And I don't think `channelIndex` can ever actually be larger than the size of `channelList` can it? It either comes from the `indexOf` call above or selection in the dropdown

**Review by jeradM — CHANGES_REQUESTED** [review]

File: src/dialogs/more-info/controls/more-info-media_player.js (hunk 1 above)
Comment (stephanerosi) [reply src/dialogs/more-info/controls/more-info-media_player.js]: I've just followed the way handleSourceChange was written. I supposed there was a reason for this (mayabe if the channel list changes and a channel is added?). If not, I can rewrite both functions.

Comment (stephanerosi) [reply src/dialogs/more-info/controls/more-info-media_player.js]: Hello, I've just corrected the length and added a comment to explain why the test is there

**Review by jeradM — APPROVED** [review]

**Comment by balloob** [comment]
Closing this as parent PR has been closed.

**Pull request closed without merging**